Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -170,22 +68,15 @@ public final class PostgresConnection: @unchecked Sendable {
     func start(configuration: InternalConfiguration) -> EventLoopFuture<Void> {
         // 1. configure handlers
 
-        var configureSSLCallback: ((Channel) throws -> ())? = nil
-        switch configuration.tls.base {
-        case .disable:
-            break
-
-        case .prefer(let sslContext), .require(let sslContext):
-            configureSSLCallback = { channel in
-                channel.eventLoop.assertInEventLoop()
+        let configureSSLCallback: ((Channel) throws -> ())? = configuration.tls.sslContext.map { sslContext in { channel in
+            channel.eventLoop.assertInEventLoop()
 
-                let sslHandler = try NIOSSLClientHandler(
-                    context: sslContext,
-                    serverHostname: configuration.sslServerHostname
-                )
-                try channel.pipeline.syncOperations.addHandler(sslHandler, position: .first)
-            }
-        }
+            let sslHandler = try NIOSSLClientHandler(
+                context: sslContext,
+                serverHostname: configuration.serverNameForSNI
+            )
+            try channel.pipeline.syncOperations.addHandler(sslHandler, position: .first)
+        } }
```

Comment:
can we keep the impl here as is? I actually like that we use the internal enum here. This will make us think about it, if we add another case in the future. Also I find the create closure inside a closure extremely hard to read.